File: lib/ts/src/internal/types/UNumber.ts
import { ValidationFailure } from '../../internal/validation/ValidationFailure';
import { UType } from '../../internal/types/UType';
import { validateNumber } from '../../internal/validation/ValidateNumber';
import { generateRandomNumber } from '../../internal/generation/GenerateRandomNumber';
import { GenerateContext } from '../../internal/generation/GenerateContext';
import { UTypeDeclaration } from '../../internal/types/UTypeDeclaration';
import { ValidateContext } from '../validation/ValidateContext';

export const numberName: string = 'Number';

export class UNumber extends UType {
    getTypeParameterCount(): number {
        return 0;
    }

    validate(value: any, typeParameters: UTypeDeclaration[], ctx: ValidateContext): ValidationFailure[] {
        return validateNumber(value);
    }

    generateRandomValue(ctx: GenerateContext): any {
        return generateRandomNumber(ctx);
    }

    getName(): string {
        return numberName;
    }
}

File: tool/uapicodegen/uapicodegen/uapi/internal/types/UNumber.py
from typing import TYPE_CHECKING


if TYPE_CHECKING:
    from ...internal.validation.ValidateContext import ValidateContext
    from ...RandomGenerator import RandomGenerator
    from ...internal.types.UTypeDeclaration import UTypeDeclaration
    from ...internal.validation.ValidationFailure import ValidationFailure
    from ...internal.generation.GenerateContext import GenerateContext


from ...internal.types.UType import UType

_NUMBER_NAME: str = "Number"


class UNumber(UType):

    def get_type_parameter_count(self) -> int:
        return 0

    def validate(self, value: object,
                 type_parameters: list['UTypeDeclaration'], ctx: 'ValidateContext') -> list['ValidationFailure']:
        from ...internal.validation.ValidateNumber import validate_number
        return validate_number(value)

    def generate_random_value(self, blueprint_value: object, use_blueprint_value: bool, type_parameters: list['UTypeDeclaration'], ctx: 'GenerateContext') -> object:
        from ...internal.generation.GenerateRandomNumber import generate_random_number
        return generate_random_number(blueprint_value, use_blueprint_value, ctx)

    def get_name(self) -> str:
        return _NUMBER_NAME
